frontend/appflowy_flutter/lib/workspace/presentation/notifications/notification_grouped_view.dart

```diff
@@ -0,0 +1,117 @@
+import 'package:appflowy/generated/locale_keys.g.dart';
+import 'package:appflowy/user/application/reminder/reminder_bloc.dart';
+import 'package:appflowy/workspace/application/appearance.dart';
+import 'package:appflowy/workspace/application/settings/date_time/date_format_ext.dart';
+import 'package:appflowy/workspace/presentation/notifications/notification_item.dart';
+import 'package:appflowy_backend/protobuf/flowy-folder2/view.pb.dart';
+import 'package:appflowy_backend/protobuf/flowy-user/reminder.pb.dart';
+import 'package:appflowy_popover/appflowy_popover.dart';
+import 'package:collection/collection.dart';
+import 'package:easy_localization/easy_localization.dart';
+import 'package:flowy_infra_ui/flowy_infra_ui.dart';
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+
+class NotificationsGroupView extends StatelessWidget {
+  const NotificationsGroupView({
+    super.key,
+    required this.groupedReminders,
+    required this.reminderBloc,
+    required this.views,
+    required this.mutex,
+    this.isUpcoming = false,
+  });
+
+  final Map<DateTime, List<ReminderPB>> groupedReminders;
+  final ReminderBloc reminderBloc;
+  final List<ViewPB> views;
+  final PopoverMutex mutex;
+  final bool isUpcoming;
+
+  @override
+  Widget build(BuildContext context) {
```

Comment:
This widget tree is too long. We are considering dividing it into smaller widgets.

Reply:
I'll abstract it further and simplify it a bit